src/EulerHS/KVConnector/Utils.hs

```diff
@@ -97,6 +103,71 @@ getDataFromRedisForPKey meshCfg pKey = do
       return $ Right Nothing
     Left e -> return $ Left $ MRedisError e
 
+groupKeysBySlot :: [ByteString] -> [[ByteString]]
```

Comment:
it's doing two traversals of the same list.

-- 
You can use `foldl'` build a hashmap (slot, [keys])